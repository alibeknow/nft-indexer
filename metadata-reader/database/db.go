package database

import (
	"context"
	"crypto/tls"
	"crypto/x509"
	"fmt"
	"os"
	"time"

	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
	"go.mongodb.org/mongo-driver/mongo/readpref"
	"go.mongodb.org/mongo-driver/mongo/writeconcern"
	"go.uber.org/zap"
)

type Config struct {
	Name             string
	Protocol         string
	Username         string
	Password         string
	Host             string
	Port             int
	SSLEnable        bool
	DirectConnection bool
}

func (config *Config) URL() string {
	url := fmt.Sprintf(
		"%s://%s:%s@%s:%d",
		config.Protocol,
		config.Username,
		config.Password,
		config.Host,
		config.Port,
	)

	if config.DirectConnection {
		url = fmt.Sprintf(
			"%s/?connect=direct",
			url,
		)
	}
	return url
}

type DB struct {
	log    *zap.Logger
	client *mongo.Client
	name   string
}

func Open(log *zap.Logger, ctx context.Context, config Config) (*DB, error) {
	opts := options.Client().ApplyURI(config.URL())

	if config.SSLEnable {
		path, _ := os.Getwd()
		pemPath := fmt.Sprintf("%s/../rds-combined-ca-bundle.pem", path)
		rootPEM, err := os.ReadFile(pemPath)
		if err != nil {
			return nil, err
		}

		roots := x509.NewCertPool()
		if ok := roots.AppendCertsFromPEM(rootPEM); !ok {
			return nil, fmt.Errorf("get certs from %s fail!\n", pemPath)
		}
		tlsConfig := &tls.Config{
			RootCAs:            roots,
			InsecureSkipVerify: true,
		}

		opts.SetReadPreference(readpref.Secondary())
		opts.SetWriteConcern(writeconcern.New(writeconcern.WMajority(), writeconcern.J(true), writeconcern.WTimeout(1000)))
		opts.SetTLSConfig(tlsConfig)
	}

	client, err := mongo.NewClient(opts)
	if err != nil {
		return nil, err
	}

	err = client.Connect(ctx)
	if err != nil {
		return nil, err
	}

	pingCtx, pingCancel := context.WithTimeout(ctx, 5*time.Second)
	defer pingCancel()
	err = client.Ping(pingCtx, nil)
	if err != nil {
		return nil, err
	}
	log.Info("Connected to DB", zap.String("url", config.URL()))

	return &DB{
		log:    log,
		client: client,
		name:   config.Name,
	}, nil
}

func (db *DB) Close() error {
	return db.client.Disconnect(context.Background())
}

func (db *DB) Metadata() *MetadataDB {
	coll := db.client.Database(db.name).Collection("metadata")
	return &MetadataDB{
		collection: coll,
	}
}
