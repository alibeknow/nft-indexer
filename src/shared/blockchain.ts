export enum Blockchain {
  ETH = 'eth',
  POLYGON = 'polygon',
}
