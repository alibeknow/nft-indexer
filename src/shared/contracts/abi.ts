export const ERC_721_TRANSFER_SIG = 'event Transfer(address indexed from, address indexed to, uint256 indexed tokenId)';
export const PRE_ERC_721_ERC_20_SIG = 'event Transfer(address indexed from, address indexed to, uint256 tokenId)';
export const PRE_ERC_721_WITHOUT_INDEXED_SIG = 'event Transfer(address from, address to, uint256 tokenId)';
export const ERC_1155_TRANSFER_SINGLE_SIG = 'event TransferSingle(address indexed operator, address indexed from, address indexed to, uint256 id, uint256 value)';
export const ERC_1155_TRANSFER_BATCH_SIG = 'event TransferBatch(address indexed operator, address indexed from, address indexed to, uint256[] ids, uint256[] values)';
