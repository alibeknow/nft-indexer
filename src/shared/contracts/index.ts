export * from './repo';
export * from './abi';
export * from './contract-search';
export * from './interfaces';
