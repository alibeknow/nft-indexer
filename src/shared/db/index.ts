export * from './constants';
export * from './db.class';
export * from './db.deprecated';
export * from './db.config';
export * from './db.module';
export * from './db.provider';
export * from './mongo-sort';

