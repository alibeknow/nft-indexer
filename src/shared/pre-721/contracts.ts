import {
  ERC_1155_TRANSFER_BATCH_SIG,
  ERC_1155_TRANSFER_SINGLE_SIG,
  ERC_721_TRANSFER_SIG,
  PRE_ERC_721_ERC_20_SIG,
  PRE_ERC_721_WITHOUT_INDEXED_SIG,
} from '@shared/contracts';
import { ContractObject } from './types';

export const CONTRACT_OBJECTS: ContractObject[] = [
  {
    // Autoglyphs
    address: '0xd4e4078ca3495DE5B1d4dB434BEbc5a986197782',
    abi: [ ERC_721_TRANSFER_SIG ],
  },
  {
    // Axie
    address: '0xF5b0A3eFB8e8E4c201e2A935F110eAaF3FFEcb8d',
    abi: [ PRE_ERC_721_ERC_20_SIG ],
  },
  {
    // Cryptocards
    address: '0x3A7DC718Eaf31f0a55988161f3d75D7CA785b034',
    abi: [ ERC_1155_TRANSFER_SINGLE_SIG, ERC_1155_TRANSFER_BATCH_SIG ],
  },
  {
    // Curio-cards
    address: '0x73DA73EF3a6982109c4d5BDb0dB9dd3E3783f313',
    abi: [ ERC_1155_TRANSFER_SINGLE_SIG, ERC_1155_TRANSFER_BATCH_SIG ],
    name: 'Curio-cards',
  },
  {
    // Decentraland
    address: '0xF87E31492Faf9A91B02Ee0dEAAd50d51d56D5d4d',
    abi: [ ERC_721_TRANSFER_SIG ],
    features: {
      api: 'https://api.decentraland.org/v2/contracts/0xf87e31492faf9a91b02ee0deaad50d51d56d5d4d/tokens/',
    },
  },
  {
    // ENS
    address: '0x57f1887a8BF19b14fC0dF6Fd9B2acc9Af147eA85',
    abi: [ ERC_721_TRANSFER_SIG ],
    name: 'ENS',
    features: {
      api: 'https://metadata.ens.domains/mainnet/0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85/',
    },
  },
  {
    // Etherbots
    address: '0xD2f81Cd7A20d60c0d558496c7169A20968389b40',
    abi: [ PRE_ERC_721_WITHOUT_INDEXED_SIG ],
    features: {
      tokenMetadata: true,
    },
  },
  {
    // Ethermon
    address: '0x5D00d312e171Be5342067c09BaE883f9Bcb2003B',
    abi: [ ERC_721_TRANSFER_SIG ],
    features: {
      api: 'https://ethermon.io/api/monster/get_metadata/v2?monster_ids=',
    },
  },
  {
    // CryptoKitties
    address: '0x06012c8cf97BEaD5deAe237070F9587f8E7A266d',
    abi: [ PRE_ERC_721_WITHOUT_INDEXED_SIG ],
    features: {
      api: 'https://api.cryptokitties.co/kitties/',
    },
  },
  {
    // PepeDapp
    address: '0xdAAe81C0077e8917a2Eb63BB66EF701ff4781Bb0',
    abi: [ ERC_1155_TRANSFER_SINGLE_SIG, ERC_1155_TRANSFER_BATCH_SIG ],
    name: 'PepeDapp',
  },
  {
    // EtherLambos
    address: '0xda9f43015749056182352e9dC6D3eE0B6293D80A',
    abi: [ PRE_ERC_721_WITHOUT_INDEXED_SIG ],
    features: {
      api: 'https://www.etherlambos.io/API/lambo/id/',
    },
  },
];
