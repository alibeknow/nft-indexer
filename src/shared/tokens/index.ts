export * from './repo';
export * from './to-valid-hex';
export * from './token-metadata-status';
export * from './token-standard';
