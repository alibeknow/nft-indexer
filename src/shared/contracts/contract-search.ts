import { Blockchain } from '@shared/blockchain';
import { TokenStandard } from '@shared/tokens';
import { IContractSearchResult } from './interfaces';
import { Contract } from './repo';

export interface IContractIndex {
  id?: string;
  address: string;
  name: string;
  description: string;
  chain: Blockchain;
  verified: boolean;
  contract_type: TokenStandard;
  number_of_tokens: number;
  block: number;
  collection_image: string;
}

export const ContractMappings = {
  properties: {
    address: {
      type: 'keyword',
      index: true,
    },
    name: {
      type: 'text',
      index: true,
    },
    description: {
      type: 'text',
      index: true,
    },
    chain: {
      type: 'keyword',
      index: true,
    },
    verified: {
      type: 'boolean',
      index: true,
    },
    contract_type: {
      type: 'text',
      index: false,
    },
    number_of_tokens: {
      type: 'integer',
      index: false,
    },
    block: {
      type: 'integer',
      index: false,
    },
    created_at: {
      type: 'text',
      index: false,
    },
    collection_image: {
      type: 'text',
      index: false,
    },
  },
};

export const mapToSearchContractIndex = (contract: Contract, tokensCount: number): IContractIndex | null => {
  try {
    const chain = contract._id.split(':')[0] as Blockchain;
    // FIXME: This is a mock
    const description = '';
    // FIXME: This is a mock
    const verified = false;

    return {
      address: contract.address,
      name: contract.name as string,
      description,
      chain,
      verified,
      contract_type: contract.type,
      number_of_tokens: tokensCount ?? 0,
      block: contract.block,
      collection_image: contract.collectionImage?.url || '',
    };
  } catch (error: unknown) {
    return null;
  }
};

export const mapToSearchContractResult = (contractData: IContractIndex): IContractSearchResult | null => {

  try {
    if (!contractData.id) {
      throw new Error('Contract id is not defined');
    }

    return {
      contract_address: contractData.address,
      contract_name: contractData.name,
      contract_description: contractData.description,
      verified: contractData.verified,
      number_of_tokens: contractData.number_of_tokens,
      contract_type: contractData.contract_type,
      synced_at: null,
      created_at: null,
      collection_image: contractData.collection_image,
    };
  } catch (error: unknown) {
    return null;
  }
};
