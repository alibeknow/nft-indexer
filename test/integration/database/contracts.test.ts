import { DBClass, IDatabaseConfig, getDatabaseConfig } from '@shared/db';
import { ContractsRepo } from '@shared/contracts';
import { Blockchain } from '@shared/blockchain';
import { TokenStandard } from '@shared/tokens';
import { contractsFixtures } from '../fixtures/contracts.fixtures';

describe('Contracts repo', () => {
  let db: DBClass;

  beforeAll(async () => {
    const dbConfig: IDatabaseConfig = getDatabaseConfig();
    db = new DBClass({ ...dbConfig, db: `db-contracts-repo-test-${dbConfig.db}` });
    await db.open();
  });

  const contractAddress = '0xc4545a0f49e1246bdba8601fd79412d240cfc6aa';

  describe('[insertIfNone]', () => {
    it('should insert new contract into db', async () => {
      const contracts = new ContractsRepo(db);

      const result = await contracts.insertIfNone({
        blockchain: Blockchain.ETH,
        address: contractAddress,
        type: TokenStandard.ERC721,
        name: 'example',
        block: 14444,
        createdAt: new Date('2022-02-01T00:00:00'),
      });

      expect(result).toMatchSnapshot();
    });

    it('should not modify existing contract', async () => {
      const contracts = new ContractsRepo(db);

      const result1 = await contracts.insertIfNone({
        blockchain: Blockchain.ETH,
        address: contractAddress,
        type: TokenStandard.ERC721,
        name: 'example',
        block: 14445,
        createdAt: new Date('2022-02-01T00:00:00'),
      });
      const result2 = await contracts.insertIfNone({
        blockchain: Blockchain.ETH,
        address: contractAddress,
        type: TokenStandard.ERC1155,
        name: 'updated',
        block: 14446,
        createdAt: new Date('2022-02-02T00:00:00'),
      });

      const cursor = db.contracts().find({});
      const contractsList = await cursor.toArray();
      expect(result1).toMatchSnapshot();
      expect(result2).toMatchSnapshot();
      expect(contractsList).toMatchSnapshot();
    });
  });

  describe('[get]', () => {
    it('should retrieve contract by id from DB', async () => {
      const id = `${Blockchain.ETH}:${contractAddress}`;
      const contracts = new ContractsRepo(db);

      await db.contracts().insertOne({
        _id: id,
        address: contractAddress,
        type: TokenStandard.ERC721,
        name: 'example',
        block: 14444,
        createdAt: new Date('2022-02-01T00:00:00'),
      });

      const contract = await contracts.get(id);
      expect(contract).toMatchSnapshot();
    });

    it('should retrieve contract by id from DB in mix lower upperCase', async () => {
      const id = `${Blockchain.ETH}:${contractAddress}`;
      const contracts = new ContractsRepo(db);

      await db.contracts().insertOne({
        _id: id,
        address: contractAddress,
        type: TokenStandard.ERC721,
        name: 'example',
        block: 14444,
        createdAt: new Date('2022-02-01T00:00:00'),
      });

      const contract = await contracts.get(id.toUpperCase());
      const contract1 = await contracts.get(id.toLowerCase());
      expect(contract).toMatchSnapshot();
      expect(contract1).toMatchSnapshot();
    });
  });

  describe('[updateName]', () => {
    it('should update name for contract stored in db', async () => {
      const id = `${Blockchain.ETH}:${contractAddress}`;
      const contracts = new ContractsRepo(db);

      await db.contracts().insertOne({
        _id: id,
        address: contractAddress,
        type: TokenStandard.ERC721,
        name: 'example',
        block: 14444,
        createdAt: new Date('2022-02-01T00:00:00'),
      });

      const result = await contracts.updateName(id, 'new name');
      const updatedContact = await db.contracts().find({ _id: id }).project({ updatedAt: 0 }).toArray();
      expect(result).toMatchSnapshot();
      expect(updatedContact[0]).toMatchSnapshot();
    });
  });

  describe('[getCreatedAfter] method', () => {
    it('should retrieve contracts documents created after a specific date', async () => {
      const contractsRepo = new ContractsRepo(db);
      await Promise.all(contractsFixtures.map(
        (contract) => contractsRepo.insertIfNone(contract),
      ));

      const result = await Promise.all([
        contractsRepo.getCreatedAfter(new Date('2019-12-17T00:00:00'), 10),
        contractsRepo.getCreatedAfter(new Date('2019-12-17T04:00:00'), 10),
        contractsRepo.getCreatedAfter(new Date('2018-12-17T04:00:00'), 10),
        contractsRepo.getCreatedAfter(new Date('2019-12-16T02:22:00'), 10),
        contractsRepo.getCreatedAfter(new Date('2019-12-17T03:26:00'), 10),
      ]);

      expect(result[0]).toHaveLength(3);
      expect(result[0]).toMatchSnapshot();
      expect(result[1]).toHaveLength(0);
      expect(result[1]).toMatchSnapshot();
      expect(result[2]).toHaveLength(5);
      expect(result[2]).toMatchSnapshot();
      expect(result[3]).toHaveLength(3);
      expect(result[3]).toMatchSnapshot();
      expect(result[4]).toHaveLength(0);
      expect(result[4]).toMatchSnapshot();
    });
  });

  afterEach(async () => {
    await db.contracts().deleteMany({});
  });

  afterAll(async () => {
    await db.close(true);
  });
});
