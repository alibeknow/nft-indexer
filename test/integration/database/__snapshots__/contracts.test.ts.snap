// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contracts repo [get] should retrieve contract by id from DB 1`] = `
{
  "_id": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "address": "0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "block": 14444,
  "createdAt": 2022-02-01T00:00:00.000Z,
  "name": "example",
  "type": "ERC-721",
}
`;

exports[`Contracts repo [get] should retrieve contract by id from DB in mix lower upperCase 1`] = `
{
  "_id": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "address": "0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "block": 14444,
  "createdAt": 2022-02-01T00:00:00.000Z,
  "name": "example",
  "type": "ERC-721",
}
`;

exports[`Contracts repo [get] should retrieve contract by id from DB in mix lower upperCase 2`] = `
{
  "_id": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "address": "0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "block": 14444,
  "createdAt": 2022-02-01T00:00:00.000Z,
  "name": "example",
  "type": "ERC-721",
}
`;

exports[`Contracts repo [getCreatedAfter] method should retrieve contracts documents created after a specific date 1`] = `
[
  {
    "_id": "eth:0x20Cc3Bd22144209CF93dA14017C506b0ed0F3e4c",
    "address": "0x20Cc3Bd22144209CF93dA14017C506b0ed0F3e4c",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:24:00.000Z,
    "name": "Audiovisions",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xC36442b4a4522E871399CD717aBDD847Ab11FE88",
    "address": "0xC36442b4a4522E871399CD717aBDD847Ab11FE88",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:25:00.000Z,
    "name": "Uniswap V3 Positions NFT-V1",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xc5846767a9c0D5096D7b791De56b258Bc7C69C5f",
    "address": "0xc5846767a9c0D5096D7b791De56b258Bc7C69C5f",
    "block": 14316462,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:26:00.000Z,
    "name": "WEEDGANG.GAME - STRAIN",
    "type": "ERC-721",
  },
]
`;

exports[`Contracts repo [getCreatedAfter] method should retrieve contracts documents created after a specific date 2`] = `[]`;

exports[`Contracts repo [getCreatedAfter] method should retrieve contracts documents created after a specific date 3`] = `
[
  {
    "_id": "eth:0x9FB2EEb75754815c5Cc9092Cd53549cEa5dc404f",
    "address": "0x9FB2EEb75754815c5Cc9092Cd53549cEa5dc404f",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-15T04:55:00.000Z,
    "name": "Superlative Mutated Apes",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xE1E484251ffFee048A839bc2d68C160BD8C82aBB",
    "address": "0xE1E484251ffFee048A839bc2d68C160BD8C82aBB",
    "block": 14316462,
    "collectionImage": null,
    "createdAt": 2019-12-16T02:22:00.000Z,
    "name": "RichApesClub",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0x20Cc3Bd22144209CF93dA14017C506b0ed0F3e4c",
    "address": "0x20Cc3Bd22144209CF93dA14017C506b0ed0F3e4c",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:24:00.000Z,
    "name": "Audiovisions",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xC36442b4a4522E871399CD717aBDD847Ab11FE88",
    "address": "0xC36442b4a4522E871399CD717aBDD847Ab11FE88",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:25:00.000Z,
    "name": "Uniswap V3 Positions NFT-V1",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xc5846767a9c0D5096D7b791De56b258Bc7C69C5f",
    "address": "0xc5846767a9c0D5096D7b791De56b258Bc7C69C5f",
    "block": 14316462,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:26:00.000Z,
    "name": "WEEDGANG.GAME - STRAIN",
    "type": "ERC-721",
  },
]
`;

exports[`Contracts repo [getCreatedAfter] method should retrieve contracts documents created after a specific date 4`] = `
[
  {
    "_id": "eth:0x20Cc3Bd22144209CF93dA14017C506b0ed0F3e4c",
    "address": "0x20Cc3Bd22144209CF93dA14017C506b0ed0F3e4c",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:24:00.000Z,
    "name": "Audiovisions",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xC36442b4a4522E871399CD717aBDD847Ab11FE88",
    "address": "0xC36442b4a4522E871399CD717aBDD847Ab11FE88",
    "block": 14316461,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:25:00.000Z,
    "name": "Uniswap V3 Positions NFT-V1",
    "type": "ERC-721",
  },
  {
    "_id": "eth:0xc5846767a9c0D5096D7b791De56b258Bc7C69C5f",
    "address": "0xc5846767a9c0D5096D7b791De56b258Bc7C69C5f",
    "block": 14316462,
    "collectionImage": null,
    "createdAt": 2019-12-17T03:26:00.000Z,
    "name": "WEEDGANG.GAME - STRAIN",
    "type": "ERC-721",
  },
]
`;

exports[`Contracts repo [getCreatedAfter] method should retrieve contracts documents created after a specific date 5`] = `[]`;

exports[`Contracts repo [insertIfNone] should insert new contract into db 1`] = `
{
  "acknowledged": true,
  "matchedCount": 0,
  "modifiedCount": 0,
  "upsertedCount": 1,
  "upsertedId": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
}
`;

exports[`Contracts repo [insertIfNone] should not modify existing contract 1`] = `
{
  "acknowledged": true,
  "matchedCount": 0,
  "modifiedCount": 0,
  "upsertedCount": 1,
  "upsertedId": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
}
`;

exports[`Contracts repo [insertIfNone] should not modify existing contract 2`] = `
{
  "acknowledged": true,
  "matchedCount": 1,
  "modifiedCount": 0,
  "upsertedCount": 0,
  "upsertedId": null,
}
`;

exports[`Contracts repo [insertIfNone] should not modify existing contract 3`] = `
[
  {
    "_id": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
    "address": "0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
    "block": 14445,
    "collectionImage": null,
    "createdAt": 2022-02-01T00:00:00.000Z,
    "name": "example",
    "type": "ERC-721",
  },
]
`;

exports[`Contracts repo [updateName] should update name for contract stored in db 1`] = `
{
  "acknowledged": true,
  "matchedCount": 1,
  "modifiedCount": 1,
  "upsertedCount": 0,
  "upsertedId": null,
}
`;

exports[`Contracts repo [updateName] should update name for contract stored in db 2`] = `
{
  "_id": "eth:0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "address": "0xc4545a0f49e1246bdba8601fd79412d240cfc6aa",
  "block": 14444,
  "createdAt": 2022-02-01T00:00:00.000Z,
  "name": "new name",
  "type": "ERC-721",
}
`;
